pkg/scanners/clairv4/clairv4.go

```diff
@@ -97,15 +114,177 @@ func validate(cfg *storage.ClairV4Config) error {
 }
 
 func (c *clairv4) GetScan(image *storage.Image) (*storage.ImageScan, error) {
-	panic("Unimplemented")
+	// For logging/error message purposes.
+	imgName := image.GetName().GetFullName()
+
+	if image.GetMetadata() == nil {
+		return nil, errors.Errorf("Clair v4: Unable to complete scan of image %s because it is missing metadata", imgName)
+	}
+
+	// Use claircore.ParseDigest instead of types.Digest (see pkg/images/types/digest.go)
+	// to mirror clairctl (https://github.com/quay/clair/blob/v4.5.0/cmd/clairctl/report.go#L251).
+	ccDigest, err := claircore.ParseDigest(imageutils.GetSHA(image))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Clair v4: parsing image digest for image %s", imgName)
+	}
+	digest := ccDigest.String()
+
+	exists, err := c.indexReportExists(digest)
+	// Exit early if this is an unexpected status code error.
+	// If it's not an unexpected error, then continue as normal and ignore the error.
+	if isUnexpectedStatusCodeError(err) {
+		log.Debugf("Clair v4: Received unexpected status from Clair: %v", err)
```

Comment:
This will logged only if the `isUnexpectedStatusCodeError(err)`. I think we should debug log the error is all cases.